DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -203,6 +211,15 @@ private void RegisterDelegate(DelegateTableRow delegateRow, DateTime? welcomeEma
                     );
                 default:
                     var newDelegateRecord = userDataService.GetDelegateUserByCandidateNumber(errorCodeOrCandidateNumber, centreId)!;
+                    if (welcomeEmailDate.HasValue)

```

Comment:
I think it might be best to place this If statement below the SetUpSupervisorDelegateRelations call so that we don't set the email up until the entire process has been completed.